File: backend/BudgetToWealthApi/Models/CashFlow/Budget.cs
public class Budget : BaseEntity
{
    public required long Amount { get; set; }
    public required Guid CategoryId { get; set; }
    public CashFlowCategory? Category { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public string? UserId { get; set; }
}

File: frontend/src/app/lib/models/cashflow/Budget.ts
import { ListTableItem } from "@/app/ui/components/table/ListTable";
import { CashFlowCategory } from "./CashFlowCategory";

export type Budget = ListTableItem & {
  id?: number;
  amount: number;
  categoryId: string;
  category?: CashFlowCategory;
  name?: string;
  startDate?: string;
  endDate?: string;
  userId?: string;
}